### DISCUSSION TITLE
Get measuring details

### DISCUSSION THREAD

**Discussion by salah-walid** [iTwin/itwinjs-core#4807] [Q&A] [answered]
Hello ITwin Team,

Using the [MeasureDistanceTool](https://www.itwinjs.org/reference/core-frontend/measure/measuredistancetool/) or [MeasureAreaTool](https://www.itwinjs.org/reference/core-frontend/measure/measureareatool/), How can we listen to measuring events and display the appropriate details in a custom component, similar to the default measuring panel shown here:

![image](https://user-images.githubusercontent.com/29980658/207837411-b80bb085-dade-4e9f-9682-27dd61cb341e.png)

**Comment by pmconne** [comment]
What measuring events are you referring to? The  tools don't define any events.
The tools are extensible. Consider subclassing them to add your custom logic.

**Comment by aruniverse** [comment, marked as answer]
Are you using measure tools from @itwin/core-frontend or @itwin/measure-tools-react? 

cc @bsy-nicholasw @a-gagnon

**Reply by salah-walid** [reply]
Using the ones from @itwin/core-frontend

Also i have noticed that tools from @itwin/measure-tools-react can be use without including the @itwin/measure-tools-react, for example use the MeasureAreaTool by running the tool id "MeasureArea", is my assumption correct ?

**Reply by aruniverse** [reply]
> tools from @itwin/measure-tools-react can be use without including the @itwin/measure-tools-react

I think you meant tools from itwin/core-frontend w/o itwin/measure-tools-react, because the `MeasureArea` tool id belongs to the tool from core-frontend.

If you want to use the ones from core-frontend you will need to extend them to add events like Paul has mentioned above.

measure-tools-react has their own implementation of some of the tools and has their own events you can listen to, [see](https://github.com/iTwin/viewer-components-react/blob/master/packages/itwin/measure-tools/src/api/MeasurementUIEvents.ts)
